src/Controller/QueueController.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/*
+ * This file is part of pmg/pheanstalk-bundle
+ *
+ * Copyright (c) PMG <https://www.pmg.com>
+ *
+ * For full copyright information see the LICENSE file distributed
+ * with this source code.
+ *
+ * @license     http://opensource.org/licenses/Apache-2.0 Apache-2.0
+ */
+
+namespace PMG\PheanstalkBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use PMG\PheanstalkBundle\Controller\Exception\InvalidTube;
+
+class QueueController extends AbstractController
+{
+    public function listTubesAction(Request $r)
+    {
+        return $this->toResponse($this->getConnection($r)->listTubes());
+    }
+
+    public function getInfoAction($tube, Request $r)
+    {
+        $conn = $this->getConnection($r);
+        $tubes = $conn->listTubes();
+
+        if (!in_array($tube, $tubes)) {
+            throw new InvalidTube(sprintf('%s is not a valid tube', $tube));
+        }
```

Comment:
Just ask for stats here, don't bother checking the tube. Pheanstalk will throw a [ServerException](https://github.com/pda/pheanstalk/blob/f095be7568f3438cd5e4d1e59016857bbcb869b1/src/YamlResponseParser.php#L44) if the tube doesn't exist.

```php
try {
  return $this->toResponse($this->getConnection($r)->statsTube($tube);
} catch (\Pheanstalk\Exception\ServerException $e) {
  throw new NotFoundHttpException(sprintf('Tube "%s" Not Found', $tube), $e);
}
```